src/box/lua/net_box.lua

```diff
@@ -390,6 +391,24 @@ local function new_sm(uri, opts)
                     })
                 end
             end
+            if not opts.fetch_schema and features.space_and_index_names then
+                remote.space = setmetatable({_remote = remote}, {
+                    __index = function(self, key)
+                        local space = {id = key, name = key,
```

Comment:
`id` is supposed to be a number while `name` should be a string. I don't think we should set both to the given key. We should set either `id` or `name` depending on the key type. The same comment for index id/name.

Reply:
Fixed.